docs/cluster-agent/HORIZONTAL_POD_AUTOSCALING.md

```diff
@@ -0,0 +1,292 @@
+
+Will yield:
+
+```
+kubectl get secret datadog-auth-token
+NAME                 TYPE      DATA      AGE
+datadog-auth-token   Opaque    1         16s
+
+```
+### Spinning up the Datadog Cluster Agent
+
+In order to spin up the Datadog Cluster Agent, you will need to create the appropriate RBAC rules.
+The DCA is acting as a proxy between the API Server and the node agent, to this extent it will need to have access to the cluster level resources.
+
+`kubectl apply -f manifests/cluster-agent/rbac-cluster-agent.yaml`
+
+```
+clusterrole.rbac.authorization.k8s.io "dca" created
+clusterrolebinding.rbac.authorization.k8s.io "dca" created
+serviceaccount "dca" created
+```
+
+Then you will need to create the DCA and its services.
+Start by adding your API_KEY and APP_KEY in the deployment manifest of the DCA.
+Then enable the HPA Processing by setting the `DD_ENABLE_HPA` vairable to true.
+Finally, spin up the resource:
+
+`kubectl apply -f manifests/cluster-agent/cluster-agent.yaml`
+`kubectl apply -f manifests/cluster-agent/datadog-cluster-agent_service.yaml`
+`kubectl apply -f manifests/cluster-agent/hpa-example/cluster-agent-hpa-svc.yaml`
+
+Note that the first service is used for the communication between the node agents and the DCA but the second is used by Kubernetes to register the External Metrics Provider.
+
+At this point you should be having:
+
+```
+PODS:
+
+NAMESPACE     NAME                                     READY     STATUS    RESTARTS   AGE
+default       datadog-cluster-agent-7b7f6d5547-cmdtc   1/1       Running   0          28m
+
+SVCS:
+
+NAMESPACE     NAME                  TYPE        CLUSTER-IP        EXTERNAL-IP   PORT(S)         AGE
+default       datadog-cluster-hpa   ClusterIP   192.168.254.87    <none>        443/TCP         28m
+default       dca                   ClusterIP   192.168.254.197   <none>        5005/TCP        28m
+
+```
+
+### Register the External Metrics Provider
+
+Once the DCA is up and running, you can register it as an External Metrics Provider, via the service exposing the port 443.
+
+To do so, simply apply the following RBAC rules:
+
+`kubectl apply -f manifest/hpa-example/rbac-hpa.yaml`
+
+```
+clusterrolebinding.rbac.authorization.k8s.io "system:auth-delegator" created
+rolebinding.rbac.authorization.k8s.io "dca" created
+apiservice.apiregistration.k8s.io "v1beta1.external.metrics.k8s.io" created
+clusterrole.rbac.authorization.k8s.io "external-metrics-reader" created
+clusterrolebinding.rbac.authorization.k8s.io "external-metrics-reader" created
+```
+ 
+Once you have the DCA running and the service registered, you can create an HPA manifest and let the DCA pull metrics from Datadog.
+The following part of the walk through explains how you can set up your agents in order to collect metrics from the applications running on your cluster.
+Those metrics will then be available for you to autoscale, via the DCA, the resources of your cluster.
```

Comment:
"Those metrics will then be available for you to autoscale the resources of your cluster, via the DCA."